packages/grid/_modules_/grid/models/rows.ts

```diff
@@ -7,6 +7,11 @@ export interface RowModel extends ObjectWithId {
   [key: string]: any;
 }
 
+export type GridUpdateAction = 'delete' | 'upsert';
+export interface RowModelUpdate extends RowModel {
+  action?: GridUpdateAction;
```

Comment:
Did you consider prefixing the prop? For instance:

```suggestion
  muiAction?: GridUpdateAction;
```

or 
```suggestion
  _action?: GridUpdateAction;
```

I wonder if we should worry about developers having a property that matches.

Reply:
The API of https://ag-grid.com/documentation/javascript/server-side-model-transactions/ completely solves this problem. I wonder if it's not better, even if it requires a few transformations on the developer's side. It also  includes features like `addIndex`, `route`, `status`

Reply:
Maybe we should think of something more complex when we have more features such as row grouping, but for the first version I suggest we stick to the scope of #935.

Reply:
We need a convention for special `id` and `action` now? (id as mentioned in #972 )
Or we could change the api to have `action` outside the row data.

Reply:
This API of ag-Grid is used for streaming, so I believe it's still in the scope of the problem we are trying to solve. I would vote for `_action` as it reminds me the private fields of http://knexjs.org/.

Reply:
> We need a convention for special id and action now? (id as mentioned in #972 )

It seems to be two problems we can solve independently. One is about how we store the data in the grid, the other about how developers input it.

Reply:
> It seems to be two problems we can solve independently. One is about how we store the data in the grid, the other about how developers input it.

Well yes we will do them separately, I was thinking of the convention for internal fields...